suite.go

```diff
@@ -109,87 +109,87 @@ func (s *suite) runStep(ctx context.Context, pickle *Scenario, step *Step, prevS
-	ctx, err = s.runBeforeStepHooks(ctx, step, err)
+	ctx, err = suite.runBeforeStepHooks(ctx, step, err)
 
-	match = s.matchStep(step)
-	s.storage.MustInsertStepDefintionMatch(step.AstNodeIds[0], match)
-	s.fmt.Defined(pickle, step, match.GetInternalStepDefinition())
+	stepDef = suite.matchStep(step)
+	suite.storage.MustInsertStepDefintionMatch(step.AstNodeIds[0], stepDef)
+	suite.fmt.Defined(pickle, step, stepDef.GetInternalStepDefinition())
 
 	if err != nil {
-		sr = models.NewStepResult(pickle.Id, step.Id, match)
-		sr.Status = models.Failed
-		s.storage.MustInsertPickleStepResult(sr)
+		stepResult = models.NewStepResult(pickle.Id, step.Id, stepDef)
+		stepResult.Status = models.Failed
+		suite.storage.MustInsertPickleStepResult(stepResult)
 
 		return ctx, err
 	}
 
-	if ctx, undef, err := s.maybeUndefined(ctx, step.Text, step.Argument); err != nil {
+	if ctx, undef, err := suite.maybeUndefined(ctx, step.Text, step.Argument); err != nil {
 		return ctx, err
 	} else if len(undef) > 0 {
-		if match != nil {
-			match = &models.StepDefinition{
+		if stepDef != nil {
+			stepDef = &models.StepDefinition{
 				StepDefinition: formatters.StepDefinition{
-					Expr:    match.Expr,
-					Handler: match.Handler,
+					Expr:    stepDef.Expr,
+					Handler: stepDef.Handler,
 				},
-				Args:         match.Args,
-				HandlerValue: match.HandlerValue,
-				Nested:       match.Nested,
+				Args:         stepDef.Args,
+				HandlerValue: stepDef.HandlerValue,
+				Nested:       stepDef.Nested,
 				Undefined:    undef,
 			}
 		}
 
-		sr = models.NewStepResult(pickle.Id, step.Id, match)
-		sr.Status = models.Undefined
-		s.storage.MustInsertPickleStepResult(sr)
+		stepResult = models.NewStepResult(pickle.Id, step.Id, stepDef)
+		stepResult.Status = models.Undefined
+		suite.storage.MustInsertPickleStepResult(stepResult)
 
-		s.fmt.Undefined(pickle, step, match.GetInternalStepDefinition())
+		suite.fmt.Undefined(pickle, step, stepDef.GetInternalStepDefinition())
 		return ctx, ErrUndefined
 	}
 
 	if prevStepErr != nil {
-		sr = models.NewStepResult(pickle.Id, step.Id, match)
-		sr.Status = models.Skipped
-		s.storage.MustInsertPickleStepResult(sr)
+		stepResult = models.NewStepResult(pickle.Id, step.Id, stepDef)
+		stepResult.Status = models.Skipped
+		suite.storage.MustInsertPickleStepResult(stepResult)
 
-		s.fmt.Skipped(pickle, step, match.GetInternalStepDefinition())
+		suite.fmt.Skipped(pickle, step, stepDef.GetInternalStepDefinition())
 		return ctx, nil
 	}
 
-	ctx, err = s.maybeSubSteps(match.Run(ctx))
+	ctx, err = suite.maybeSubSteps(stepDef.Run(ctx))
 
 	return ctx, err
 }
 
-func (s *suite) runBeforeStepHooks(ctx context.Context, step *Step, err error) (context.Context, error) {
+func (suite *testSuite) runBeforeStepHooks(ctx context.Context, step *Step, err error) (context.Context, error) {
 	hooksFailed := false
 
-	for _, f := range s.beforeStepHandlers {
-		hctx, herr := f(ctx, step)
+	for _, handler := range suite.beforeStepHandlers {
```

Comment:
This particular usage like others isn't _wrong_ as it's still valid go, though it does qualify as against styling. I'd also add that `handler` is only used once within the loop on the immediately following line, could be assumed that it is a handler given we're ranging over something called `beforeStepHandlers`, and it also breaks the consistency of the following `hctx` & `herr` for handler context and handler error respectively.

Reply:
There is indeed some historical inconsistency in names related to hooks, I think it would be beneficial to use word `hook` instead of `handler` where hook semantics is implied (e.g. rename `beforeStepHandlers`). 

Range variable `f` denotes function, I think `h` might be a better name.

Reply:
One reason I wanted to use a longer name here was because I missed the most important line (where the hook function is called) initially when skimming the code.

Reply:
I've renamed it to `hook` but we can use `h` if you think that's more idiomatic.

Reply:
@vearutop I've also renamed `*handler` to `*hook` everywhere. Better?